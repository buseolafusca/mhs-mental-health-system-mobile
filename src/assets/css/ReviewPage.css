#page-container {
    position: relative;
    min-height: 100vh;
  }

.table-contain {
    padding-bottom: 0.5rem; 
}