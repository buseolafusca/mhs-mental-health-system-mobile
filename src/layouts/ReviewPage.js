import React, { Component } from "react";
import {fetchUserAnswers} from "../services/BackendService";
import $ from "jquery";
import NHSHeader from '../components/NHSHeader.js'
import NHSFooter from '../components/NHSFooter.js'
import "../assets/css/ReviewPage.css";

$(document).ready(function($) {
    $(".nhsuk-table__row").click(function() {
        window.location = $(this).data("href");
    });
});
class ReviewPage extends Component {
    constructor (props) {
      super(props)
      this.state = { userAnswers: []
      }
    }
  
    componentWillMount () {
      fetchUserAnswers()
        .then(response => {
          var rows = []
          var tableData;
          for (var i = 0; i < response.length; i++) {
            var row = [response[i].title, 'PENDING']
            rows.push(row)
            tableData += "<tr class='nhsuk-table__row' data-id=" + response[i].questionnaire_id + " >"
            tableData += "<td class='nhsuk-table__cell'>" + response[i].title + "</td>"

            tableData += "<td class='nhsuk-table__cell'>" + 'PENDING' + "</td>"
            tableData += "</tr>"
          }
          //this.setState({ userAnswers: rows })
          $("#tbody1").html(tableData);
          this.addRowHandlers();
        })
        .catch(error => {
          console.error(error)
        })
    }
  
    addRowHandlers() {
        var table = document.getElementById("tbody1");
        var rows = table.getElementsByTagName("tr");
        for (var i = 0; i < rows.length; i++) {
          var currentRow = table.rows[i];
          var createClickHandler = function(row) {
            return function() {
              console.log(currentRow.getAttribute('data-id'));
              window.location.href='http://localhost:3000/result';
            };
          };
          currentRow.onclick = createClickHandler(currentRow);
        }
      }
  
    render () {
      return ( 
        <div id = "page-container">
          <NHSHeader/>
            <div class="nhsuk-table-responsive" id = 'table-contain'>
                <table id = "get-table" class="nhsuk-table">
                    <caption class="nhsuk-table__caption">Your Questionnaire Status</caption>
                    <thead class="nhsuk-table__head">
                        <tr class="nhsuk-table__row">
                            <th class="nhsuk-table__header" scope="col">Questionnaire</th>
                            <th class="nhsuk-table__header" scope="col">Status</th>
                        </tr>
                    </thead>
                    
                    <tbody class="nhsuk-table__body" id = "tbody1">
                
                    </tbody>
                </table>
            
            </div>
            <NHSFooter/>
        </div>
      )
    }
  }

export default ReviewPage