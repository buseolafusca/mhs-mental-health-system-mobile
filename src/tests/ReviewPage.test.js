import React from 'react'
import { shallow } from 'enzyme'
import ReviewPage from '../layouts/ReviewPage'

describe('Review page', () => {
  it('should render correctly in mode', () => {
    const component = shallow(<ReviewPage/>)

    expect(component).toMatchSnapshot()
  })

  it('renders an `#get-table`', () => {
    const wrapper = shallow(<ReviewPage />);
    expect(wrapper.find('#get-table')).toHaveLength(1);
  });
})
