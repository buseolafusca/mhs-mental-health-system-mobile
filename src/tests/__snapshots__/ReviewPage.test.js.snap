// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review page should render correctly in mode 1`] = `
<div
  id="page-container"
>
  <NHSHeader />
  <div
    class="nhsuk-table-responsive"
    id="table-contain"
  >
    <table
      class="nhsuk-table"
      id="get-table"
    >
      <caption
        class="nhsuk-table__caption"
      >
        Your Questionnaire Status
      </caption>
      <thead
        class="nhsuk-table__head"
      >
        <tr
          class="nhsuk-table__row"
        >
          <th
            class="nhsuk-table__header"
            scope="col"
          >
            Questionnaire
          </th>
          <th
            class="nhsuk-table__header"
            scope="col"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody
        class="nhsuk-table__body"
        id="tbody1"
      />
    </table>
  </div>
  <NHSFooter />
</div>
`;
